core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1302,7 +1278,8 @@ class ReplicaManager(val config: KafkaConfig,
         stateChangeLogger.warn(stateControllerEpochErrorMessage)
         throw new ControllerMovedException(stateChangeLogger.messageWithPrefix(stateControllerEpochErrorMessage))
       } else {
-        val deletedPartitions = metadataCache.updateMetadata(correlationId, updateMetadataRequest)
+        val zkMetadataCache = metadataCache.asInstanceOf[ZkMetadataCache]
```

Comment:
Could we use `MetadataSupport` in `ReplicaManager`?

Reply:
That class brings in a bunch of stuff we don't need in `ReplicaManager`, though, like an `AdminZkClient` and `KafkaController`, etc...